### PR TITLE
Feat(web, web-react): `PartnerLogo` - prep for the stabilization

### PR DESCRIPTION
**Pull request by pavelklibani**
## Description
- Modifier class changed to `safeArea` from `hasSafeAreaDisabled`
- prop changed from `hasSafeAreaDisabled` -> `hasSafeArea`  and default value changed from `false` -> `true`

### Issue reference

[PartnerLogo - preparation for the stabilization](https://jira.almacareer.tech/browse/DS-1373)

### CODE REVIEW

**Review by literat — APPROVED** [review]

File: packages/web-react/src/components/UNSTABLE_PartnerLogo/__tests__/usePartnerLogoStyleProps.test.ts [hunk 1]
```diff
@@ -10,18 +10,27 @@ describe('usePartnerLogoStyleProps', () => {
...
-
-    it('should return hasSafeAreaDisabled class names', () => {
-      expect(result.current.classProps).toBe('UNSTABLE_PartnerLogo UNSTABLE_PartnerLogo--hasSafeAreaDisabled');
-    });
-
-    it('should not return hasSafeAreaDisabled prop', () => {
-      expect(result.current.props).toStrictEqual({});
+  describe('hasSafeArea prop', () => {
+    const testCases = [
+      { hasSafeArea: false, expectedClassName: 'UNSTABLE_PartnerLogo' },
```
Comment (curdaj) [inline packages/web-react/src/components/UNSTABLE_PartnerLogo/__tests__/usePartnerLogoStyleProps.test.ts:15]:
```suggestion
      { hasSafeArea: false, expectedClassName: 'UNSTABLE_PartnerLogo', description: 'should not ' },
```

File: packages/web-react/src/components/UNSTABLE_PartnerLogo/__tests__/usePartnerLogoStyleProps.test.ts [hunk 2]
```diff
@@ -10,18 +10,27 @@ describe('usePartnerLogoStyleProps', () => {
...
+      {
+        hasSafeArea: true,
+        expectedClassName: 'UNSTABLE_PartnerLogo UNSTABLE_PartnerLogo--safeArea',
+      },
```
Comment (curdaj) [inline packages/web-react/src/components/UNSTABLE_PartnerLogo/__tests__/usePartnerLogoStyleProps.test.ts:19]:
```suggestion
      description: 'should',
      },
```

File: packages/web-react/src/components/UNSTABLE_PartnerLogo/__tests__/usePartnerLogoStyleProps.test.ts [hunk 3]
```diff
@@ -10,18 +10,27 @@ describe('usePartnerLogoStyleProps', () => {
...
+    ];
+
+    testCases.forEach(({ hasSafeArea, expectedClassName }) => {
```
Comment (curdaj) [inline packages/web-react/src/components/UNSTABLE_PartnerLogo/__tests__/usePartnerLogoStyleProps.test.ts:22]: describe.each(testCases)(`when hasSafeArea is $hasSafeArea`, ({ hasSafeArea, expectedClassName, description }) => {
      const { result } = renderHook(() => usePartnerLogoStyleProps({ hasSafeArea }));

      it(`${description} return safe area class name`, () => {
        expect(result.current.classProps).toBe(expectedClassName);
      });

      it('should not return hasSafeArea prop', () => {
        expect(result.current.props).toStrictEqual({});
      });
    });
  });

**Review by curdaj — APPROVED** [review]

File: packages/web-react/src/components/UNSTABLE_PartnerLogo/stories/UNSTABLE_PartnerLogo.stories.tsx [hunk 4]
```diff
@@ -23,7 +23,7 @@ const meta: Meta<typeof UNSTABLE_PartnerLogo> = {
     },
   },
   args: {
-    hasSafeAreaDisabled: false,
+    hasSafeArea: true,
```
Comment (crishpeen) [inline packages/web-react/src/components/UNSTABLE_PartnerLogo/stories/UNSTABLE_PartnerLogo.stories.tsx:26]: this prop should be in argTypes too, please add it there. We missed that before.

File: packages/web-react/src/components/UNSTABLE_PartnerLogo/usePartnerLogoStyleProps.ts [hunk 5]
```diff
@@ -8,14 +8,14 @@ export interface PartnerLogoStyles<T> {
 }
 
 export function usePartnerLogoStyleProps(props: SpiritPartnerLogoProps): PartnerLogoStyles<SpiritPartnerLogoProps> {
-  const { size, hasSafeAreaDisabled, ...restProps } = props;
+  const { size, hasSafeArea, ...restProps } = props;
 
   const partnerLogoClass = useClassNamePrefix('UNSTABLE_PartnerLogo');
   const partnerLogoSizeClass = `${partnerLogoClass}--${size}`;
-  const partnerLogoSafeAreaDisabledClass = `${partnerLogoClass}--hasSafeAreaDisabled`;
+  const partnerLogoSafeAreaDisabledClass = `${partnerLogoClass}--safeArea`;
```
Comment (crishpeen) [inline packages/web-react/src/components/UNSTABLE_PartnerLogo/usePartnerLogoStyleProps.ts:15]:
```suggestion
  const partnerLogoSafeAreaClass = `${partnerLogoClass}--safeArea`;
```

File: packages/web-react/src/components/UNSTABLE_PartnerLogo/usePartnerLogoStyleProps.ts [hunk 6]
```diff
@@ -8,14 +8,14 @@ export interface PartnerLogoStyles<T> {
...
   const classProps = classNames(partnerLogoClass, {
     [partnerLogoSizeClass]: size,
-    [partnerLogoSafeAreaDisabledClass]: hasSafeAreaDisabled,
+    [partnerLogoSafeAreaDisabledClass]: hasSafeArea,
```
Comment (crishpeen) [inline packages/web-react/src/components/UNSTABLE_PartnerLogo/usePartnerLogoStyleProps.ts:18]:
```suggestion
    [partnerLogoSafeAreaClass]: hasSafeArea,
```

File: packages/web-react/src/types/partnerLogo.ts [hunk 7]
```diff
@@ -2,5 +2,5 @@ import { ChildrenProps, SizesDictionaryType, StyleProps } from './shared';
 
 export interface SpiritPartnerLogoProps<S = void> extends ChildrenProps, StyleProps {
   size?: SizesDictionaryType<S> | S;
-  hasSafeAreaDisabled?: boolean;
+  hasSafeArea?: boolean;
```
Comment (crishpeen) [inline packages/web-react/src/types/partnerLogo.ts:5]: abc?

**Review by crishpeen — APPROVED** [review]

**Comment by pavelklibani** [comment]
rebased to actual [integration/BC-v3-design-tokens](https://github.com/lmc-eu/spirit-design-system/tree/integration/BC-v3-design-tokens)

**Pull request merged**